internal/benchrunner/runners/system/runner.go

```diff
@@ -691,6 +708,159 @@ func (r *runner) enrollAgents() error {
 	return nil
 }
 
+func (r *runner) reindexData() error {
+	if !r.options.ReindexData {
+		return nil
+	}
+	if r.options.ESMetricsAPI == nil {
+		return errors.New("the option to reindex data is set, but the metricstore was not initialized")
+	}
+
+	logger.Debug("starting reindexing of data...")
+
+	logger.Debug("gettings orignal mappings...")
+	// Get the mapping from the source data stream
+	mappingRes, err := r.options.ESAPI.Indices.GetMapping(
+		r.options.ESAPI.Indices.GetMapping.WithIndex(r.runtimeDataStream),
+	)
+	if err != nil {
+		return fmt.Errorf("error getting mapping: %w", err)
+	}
+	defer mappingRes.Body.Close()
+
+	body, err := io.ReadAll(mappingRes.Body)
+	if err != nil {
+		return fmt.Errorf("error reading mapping body: %w", err)
+	}
+
+	mappings := map[string]struct {
+		Mappings json.RawMessage
+	}{}
+
+	if err := json.Unmarshal(body, &mappings); err != nil {
+		return fmt.Errorf("error unmarshaling mappings: %w", err)
+	}
+
+	if len(mappings) != 1 {
+		return fmt.Errorf("exactly 1 mapping was expected, got %d", len(mappings))
+	}
+
+	var mapping string
+	for _, v := range mappings {
+		mapping = string(v.Mappings)
+	}
+
+	reader := bytes.NewReader(
+		[]byte(fmt.Sprintf(`{
+			"settings": {"number_of_replicas":0},
+			"mappings": %s
+		}`, mapping)),
+	)
+
+	indexName := fmt.Sprintf("bench-reindex-%s-%s", r.runtimeDataStream, r.ctxt.Test.RunID)
+
+	logger.Debugf("creating %s index in metricstore...", indexName)
+
+	createRes, err := r.options.ESMetricsAPI.Indices.Create(
```

Comment:
How is this index different to the one created in `createMetricsIndex`? Both are created in the metrics ES.

Reply:
Metricstore by default will store metrics data from the benchmark and optionally reindexed data, to keep these 2 data types separated, we create this second index using the same mappings as the original ones from the benchmark.